Scripts/Python/kdshTreeRingsSolution.py

```diff
@@ -241,45 +241,15 @@ def OnServerInitComplete(self):
         self.InitRings()
 
     def InitRings(self):
-        ageSDL = PtGetAgeSDL()        
-        
-        OuterRing01 = ageSDL["OuterRing01"][0]
-        MiddleRing01 = ageSDL["MiddleRing01"][0]
-        InnerRing01 = ageSDL["InnerRing01"][0]
-        OuterRing02 = ageSDL["OuterRing02"][0]
-        MiddleRing02 = ageSDL["MiddleRing02"][0]
-        InnerRing02 = ageSDL["InnerRing02"][0]
-        OuterRing03 = ageSDL["OuterRing03"][0]
-        MiddleRing03 = ageSDL["MiddleRing03"][0]
-        InnerRing03 = ageSDL["InnerRing03"][0]
+        PtDebugPrint("kdshTreeRingSolution: When I got here:", level=kWarningLevel)
 
-        PtDebugPrint("kdshTreeRingSolution: When I got here:")
-        #~ PtDebugPrint("\tOuterRing01=",OuterRing01)
-        #~ PtDebugPrint("\tMiddleRing01=",MiddleRing01)
-        #~ PtDebugPrint("\tInnerRing01=",InnerRing01)
-        #~ PtDebugPrint("\tOuterRing02=",OuterRing02)
-        #~ PtDebugPrint("\tMiddleRing02=",MiddleRing02)
-        #~ PtDebugPrint("\tInnerRing02=",InnerRing02)
-        #~ PtDebugPrint("\tOuterRing03=",OuterRing03)
-        #~ PtDebugPrint("\tMiddleRing03=",MiddleRing03)
-        #~ PtDebugPrint("\tInnerRing03=",InnerRing03)
-
-        for i in ["Outer", "Middle", "Inner"]:
-            for j in ["1","2","3"]:
-                
-                ffcode1 = "InitState = "+i + "Ring0" +j
-                #~ PtDebugPrint("ffcode1 = ", ffcode1)
-                
-                exec(ffcode1)
-                #~ PtDebugPrint("InitState = ", InitState)
-                
-                ffcode2 = i + "Ring0" + j + "_0" + str(InitState) + ".animation.skipToEnd()"
-                #~ PtDebugPrint("ffcode2 = ", ffcode2)
-                
-                exec(ffcode2)
-            
-                #~ PtDebugPrint("Fastforwarding: Set = ",j," Ring = ",i," Position = ",InitState)
-                PtDebugPrint("\t",i,"Ring0",j," = ", InitState)
+        ageSDL = PtGetAgeSDL()
+        for i in ("Outer", "Middle", "Inner"):
+            for j in ("1", "2", "3"):
+                ring = "{location}Ring0{idx}".format(location=i, idx=j)
+                ring_attrib = "{ring}_0{bearing}".format(ring=ring, bearing=ageSDL[ring][0])
+                state = globals()[ring_attrib].animation.skipToEnd()
+                PtDebugPrint("{} = {}".format(ring_attrib, state), level=kWarningLevel)
```

Comment:
The old code printed the InitState (i.e. the value read from ageSDL[ring][0]), but this prints the result of `skipToEnd()`...

Also, missing leading tab (if unintentional)

Reply:
Good catch. I had a time figuring out what this particular mess of exec statements was doing.